packages/muelu/src/Misc/MueLu_MultiVectorAggregationTransferFactory_def.hpp

```diff
@@ -0,0 +1,177 @@
+    validParamList->set< RCP<const FactoryBase> >("Vector factory", Teuchos::null, "Factory of the vector");
+    validParamList->set< RCP<const FactoryBase> >("Aggregates",     Teuchos::null, "Factory of the aggregates");
+    validParamList->set< RCP<const FactoryBase> >("CoarseMap",     Teuchos::null, "Factory of the coarse map");
+
+    return validParamList;
+  }
+
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  void MultiVectorAggregationTransferFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::DeclareInput(Level &fineLevel, Level &coarseLevel) const {
+    const ParameterList & pL = GetParameterList();
+    std::string vectorName   = pL.get<std::string>("Vector name");
+
+    fineLevel.DeclareInput(vectorName, GetFactory("Vector factory").get(), this);
+    Input(fineLevel, "Aggregates");
+    Input(fineLevel, "CoarseMap");
+  }
+
+  template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  void MultiVectorAggregationTransferFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Build(Level & fineLevel, Level &coarseLevel) const {
+    FactoryMonitor m(*this, "Build", coarseLevel);
+
+    const ParameterList & pL = GetParameterList();
+    std::string vectorName   = pL.get<std::string>("Vector name");
+
+    if (vectorName == "Coordinates")
+      TEUCHOS_TEST_FOR_EXCEPTION(true,Exceptions::RuntimeError,"Use CoordinatesTransferFactory to transfer coordinates instead of MultiVectorAggregationTransferFactory.");
+
+    int numDimensions;
+    RCP<MultiVector> coarseVector;
+    RCP<MultiVector> fineVector;
+
+    if (coarseLevel.IsAvailable(vectorName, this)) {
+      GetOStream(Runtime0) << "Reusing " << vectorName << std::endl;
+      return;
+    }
+
+    RCP<Aggregates> aggregates = Get< RCP<Aggregates> > (fineLevel, "Aggregates");
+    fineVector = fineLevel.Get<RCP<MultiVector> >(vectorName);
+
+    if (aggregates->AggregatesCrossProcessors())
+      TEUCHOS_TEST_FOR_EXCEPTION(true,Exceptions::RuntimeError,"MultiVectorAggregationTransferFactory does not support aggregates that cross processors!");
+
+    // if the coarse map is available, use it. otherwise, construct the map based on aggregates
+    RCP<const Map> coarseMap = Get< RCP<const Map> > (fineLevel, "CoarseMap");
+
+    // coarseMap is being used to set up the domain map of tentative P, and therefore, the row map of Ac
+    // Therefore, if we amalgamate coarseMap, logical nodes in the coordinates vector would correspond to
+    // logical blocks in the matrix
+
+    ArrayView<const GO> elementAList = coarseMap->getLocalElementList();
+
+    LO                  blkSize      = 1;
+    if (rcp_dynamic_cast<const StridedMap>(coarseMap) != Teuchos::null)
+      blkSize = rcp_dynamic_cast<const StridedMap>(coarseMap)->getFixedBlockSize();
```

Comment:
Be careful, see PR #10829.  In particular, see [this comment](https://github.com/trilinos/Trilinos/pull/10829/files/44da6788e571d635ba45b751911d65cb0147ac7c#r939453218).  @csiefer2 Could this be a problem?

Reply:
Oh, good catch! I'll watch out for this